dynamic/main.go

```diff
@@ -62,14 +62,23 @@ func initialSetup() (info.Provider, pfbridge.ProviderMetadata, func() error) {
 	}
 
 	var metadata pfbridge.ProviderMetadata
+	var fullDocs bool
 	metadata = pfbridge.ProviderMetadata{
 		XGetSchema: func(ctx context.Context, req plugin.GetSchemaRequest) ([]byte, error) {
+			// By default, only read in docs from TF schema
+			var schemaDocsOnly = true
+			// If full docs is expected set schemaDocsOnly to false
+			if fullDocs {
+				schemaDocsOnly = false
+			}
+
 			packageSchema, err := tfgen.GenerateSchemaWithOptions(tfgen.GenerateSchemaOptions{
 				ProviderInfo: info,
 				DiagnosticsSink: diag.DefaultSink(os.Stdout, os.Stderr, diag.FormatOptions{
 					Color: colors.Always,
 				}),
-				XInMemoryDocs: true,
+				XInMemoryDocs:            schemaDocsOnly,
+				XLoadUpstreamRepoForDocs: fullDocs,
```

Comment:
Instead of keeping track of both `schemaDocsOnly` and `fullDocs` (and keeping them in sync), we can simplify to one variable.
```suggestion
				XInMemoryDocs:            !fullDocs,
				XLoadUpstreamRepoForDocs: fullDocs,
```